src/orion/core/worker/algo_wrappers/algo_wrapper.py

```diff
@@ -0,0 +1,321 @@
+            # it in the registry (for example in testing when we force the algo to observe a trial).
+            self.registry.register(trial)
+
+            # Get the known transformed trials that correspond to this original.
+            transformed_trials = self.registry_mapping.get_trials(trial)
+
+            # Also transfer the status and results of `trial` to the equivalent transformed trials.
+            transformed_trials = [
+                _copy_status_and_results(
+                    trial_with_status=trial, trial_with_params=transformed_trial
+                )
+                for transformed_trial in transformed_trials
+            ]
+
+            if not transformed_trials:
+                # `trial` is a new, original trial that wasn't suggested by the algorithm. (This
+                # might happen when an insertion is done according to @bouthilx)
+                transformed_trial = self.transform(trial)
+                transformed_trial = _copy_status_and_results(
+                    trial_with_status=trial, trial_with_params=transformed_trial
+                )
+                transformed_trials = [transformed_trial]
+                logger.debug(
+                    f"Observing trial {trial} (transformed as {transformed_trial}), even "
+                    f"though it wasn't suggested by the algorithm."
+                )
+                # NOTE: @lebrice Here we don't want to store the transformed trial in the
+                # algo's registry (by either calling `self.algorithm.register(transformed_trial)` or
+                # `self.algorithm.registry.register(transformed_trial))`, because some algos can't
+                # observe trials that they haven't suggested. We'd also need to perform all the
+                # logic that the algo did in `suggest` (e.g. store it in a bracket for HyperBand).
+                # Therefore we only register it in the wrapper, and store the equivalence between
+                # these two trials in the registry mapping.
+                self.registry.register(trial)
+                self.registry_mapping.register(trial, transformed_trial)
+
+            self.algorithm.observe(transformed_trials)
+
+    def __repr__(self) -> str:
+        return f"<{type(self).__qualname__}{self.algorithm}>"
+
+    @property
+    def is_done(self) -> bool:
+        """Return True, if an algorithm holds that there can be no further improvement."""
+        return super().is_done or self.algorithm.is_done
+
+    @property
+    def configuration(self) -> dict:
+        """Return tunable elements of this algorithm in a dictionary form
+        appropriate for saving.
+
+        Subclasses should overwrite this method and add any of their state.
+        """
+        # TODO: Do we also save the algorithm wrapper's configuration here? Or only the algo's?
+        # For now, we'll just always return the algo's config.
+        return self.algorithm.configuration
```

Comment:
If we can avoid needing to configure that wrapper that would be best.

Reply:
So we agree we can remove the TODO?

Reply:
Yeah, I made it a NOTE. Sorry I don't know which commit exactly.